custom_components/tuya_local/devices/casdon_td_pro_2.yaml

```diff
@@ -0,0 +1,206 @@
+name: Casdon TD Pro 2
+products:
+  - id: wab6oustdukujpkn
+    name: 蒸烤箱TD-pro二代
+primary_entity:
+  entity: switch
+  icon: "mdi:power"
+  name: Switch
+  dps:
+    - id: 1
+      type: boolean
+      name: switch
+
+secondary_entities:
+  - entity: switch
+    name: Start
+    icon: "mdi:play-pause"
+    dps:
+      - id: 2
+        name: switch
+        type: boolean
+  - entity: number
+    name: cook_temperature
+    dps:
+      - id: 7
+        name: value
+        type: integer
+        unit: C
+        range:
+          min: 0
+          max: 500
+  - entity: sensor
+    name: temp_current
+    class: temperature
+    dps:
+      - id: 8
+        name: sensor
+        type: integer
+        unit: C
+        readonly: true
+        range:
+          min: 0
+          max: 500
+  - entity: number
+    name: appointment_time
+    dps:
+      - id: 9
+        name: value
+        type: integer
+        optional: true
+        range:
+          min: 0
+          max: 1440
+        unit: min
+        class: measurement
+  - entity: number
+    icon: "mdi:timer-edit"
+    name: cook_time
+    dps:
+      - id: 10
+        name: value
+        type: integer
+        range:
+          min: 0
+          max: 1440
+        unit: min
+        class: measurement
+  - entity: sensor
+    icon: "mdi:timer"
+    name: remain_time
```

Comment:
Use `translation_key: time_remaining` instead of name and icon.
And add `class: duration`, for better formatting of HA display.